arrow-libs/resilience/arrow-resilience/src/commonMain/kotlin/arrow/resilience/ScheduleEither.kt

```diff
@@ -0,0 +1,30 @@
+import arrow.core.Either
+import arrow.core.raise.Raise
+import arrow.core.raise.either
+import arrow.resilience.Schedule
+import arrow.resilience.ScheduleStep
+import kotlinx.coroutines.currentCoroutineContext
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.ensureActive
+import kotlin.time.Duration
+
+public suspend fun <Error, Result, Output> Schedule<Error, Output>.retry(
+  action: suspend Raise<Error>.() -> Result,
+): Either<Error, Result> {
```

Comment:
I think this only works atm, because this is in a different package. See the test, it imports `import retry`.
That's not really desirable at all, the problem is going to be if this lands in `arrow.resilience` there won't be a way for the compiler to distinct between `retry` for `Throwable` and non-Throwable.

Perhaps by forcefully passing a generic parameter, but that's really not desired. So we'll probably have to rename `retry` to `retryRaise` but we can keep the one for `Either`, since that will overload correctly.

I am personally fine having both signatures, since I discussed with @serras some time ago about supporting `Either` in a similar way than `Raise`. That's why we also have `Either.zip` that works directly with `Either` values. That could also be an argument to keep traverse for `Either` though... 🤔 cc\\ @serras

Nonetheless, great work @akotynski. This was desperately missed for quite some time!

Reply:
Indeed, I am up for exposing both a Raise and and Either version, if possible.

About naming, we need to be conscious of our compatibility promise, so we should keep the existing function as is, and add new retryXX for the new ones.

Reply:
thanks! I renamed function and kept both signatures:
```kt
public suspend inline fun <Error, Result, Output> Schedule<Error, Output>.retryRaise(
  @BuilderInference action: Raise<Error>.() -> Result,
): Either<Error, Result>
```
```kt
public suspend inline fun <Error, Result, Output> Schedule<Error, Output>.retryEither(
  @BuilderInference action: () -> Either<Error, Result>,
): Either<Error, Result>
```

Reply:
I would suggest adding another one working completely within `Raise`:

```kotlin
public suspend inline fun <Error, Result, Output> Raise<Error<>.retryRaise(
  schedule: Schedule<Error, Output>,
  @BuilderInference action: Raise<Error>.() -> Result,
): Either<Error, Result>
```

so that then we would get:

```kotlin
public suspend inline fun <Error, Result, Output> Schedule<Error, Output>.retryRaise(
  @BuilderInference action: Raise<Error>.() -> Result,
): Either<Error, Result> = either { retyRaise(this@retryRaise) }

public suspend inline fun <Error, Result, Output> Schedule<Error, Output>.retryEither(
  @BuilderInference action: () -> Either<Error, Result>,
): Either<Error, Result> = this.retryRaise { action().bind() }
```

Reply:
ahh, yes, with having Raise in the context we do not need to return Either, we can just raise error :) 
I was just fixed on the extension function on Schedule class.
So the base signature will be:
```kt
public suspend inline fun <Error, Result, Output> Raise<Error>.retry(
  schedule: Schedule<Error, Output>,
  @BuilderInference action: Raise<Error>.() -> Result,
): Result
```